### PR TITLE
Feature/air2204 4 personal schedule

### PR DESCRIPTION
**Pull request by karlo-rusovan**
https://air2204.atlassian.net/browse/AIR2204-4

### CODE REVIEW

File: lib/Utilities/utils.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import 'package:flutter/material.dart';
+
+class Utils {
+  static final messengerKey = GlobalKey<ScaffoldMessengerState>();
```
Comment (DreamAgencyGithub) [inline lib/Utilities/utils.dart]: U PR za autentikaciju sam ovo spomenuo, koristite `ScaffoldMessenger.of(context)` umjesto keya

File: lib/application/presentation/personal_schedule.dart [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../../constants/schedule_item_categories.dart';
+import '../../domain/models/schedule_items.dart';
+import '../logic/saved_schedule_cubit.dart';
+import 'action_items.dart';
+
+class PersonalSchedule extends StatefulWidget {
```
Comment (DreamAgencyGithub) [inline lib/application/presentation/personal_schedule.dart:14]: simulator_screenshot_BC66C36C-6F61-4ABE-90AB-F1725350E00C
Brand new account, dobijem ovu grešku. Također, ispis greške i buttona bi trebalo biti centrirano. Isto vrijedi i za ostale ekrane.


Također u konzoli dobijem ovo:
`════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown while applying parent data.:
Incorrect use of ParentDataWidget.

The ParentDataWidget Expanded(flex: 1) wants to apply ParentData of type FlexParentData to a RenderObject, which has been set up to accept ParentData of incompatible type BoxParentData.

Usually, this means that the Expanded widget has the wrong ancestor RenderObjectWidget. Typically, Expanded widgets are placed directly inside Flex widgets.
The offending Expanded is currently placed inside a Center widget.

The ownership chain for the RenderObject that received the incompatible parent data was:
  Column ← Expanded ← Center ← RetryButton ← BlocListener<SavedScheduleCubit, SavedScheduleState> ← BlocBuilder<SavedScheduleCubit, SavedScheduleState> ← Column ← KeyedSubtree-[GlobalKey#7b686] ← _BodyBuilder ← MediaQuery ← ⋯
When the exception was thrown, this was the stack`

Pogledajte si RetryButton widget. Parent Expanded ili Flexible widgetu mora biti: Row, Column ili Flex widget

File: lib/application/logic/saved_schedule_state.dart [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+  const SavedScheduleState(
+      {this.scheduleItems = const [],
+      this.savedItems = const [],
+      this.loading = true,
+      this.error,
+      required this.currentDate,
+      required this.allDates,
+      this.currentCategory = ScheduleItemCategory.all});
+
```
Comment (DreamAgencyGithub) [inline lib/application/logic/saved_schedule_state.dart:20]: Dodajte si ovdje poseban konstruktor tipa `SavedScheduleState.initial()` pa onda stavite da taj konstruktor uvijek returna ono što koristite u Cubitu kao počeno stanje i što u Cubitu unutar `readAllSavedItems` metode prvo emitate.

File: lib/application/logic/saved_schedule_cubit.dart [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+      DateTime currentDate = state.currentDate;
+      List<ScheduleItem> dataFiltered = data.where((scheduleItem) => scheduleItem.date == state.currentDate).toList();
+
+      if (dataFiltered.isEmpty) {
+        dataFiltered = data.where((scheduleItem) => scheduleItem.date == allDates.first).toList();
+        currentDate = allDates.first;
+      }
+
+      emit(SavedScheduleState(
```
Comment (DreamAgencyGithub) [inline lib/application/logic/saved_schedule_cubit.dart]: image
Ako si niste stavili ovu ekstenziju za VsCode, stavite si je. Onda si uz pomoć nje izgenerirajte copyWith metodu.
I onda kod ovog emita si napravite ovako:
`emit(state.copyWith(allDates: dates, savedItems: data, scheduleItems: dataFiltered, loading: false, currentDate: currentDate));`

File: lib/application/logic/saved_schedule_cubit.dart [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+          allDates: allDates,
+          savedItems: data,
+          scheduleItems: dataFiltered,
+          loading: false,
+          currentCategory: ScheduleItemCategory.all,
+          currentDate: currentDate));
+    } catch (e) {
+      emit(SavedScheduleState(
```
Comment (DreamAgencyGithub) [inline lib/application/logic/saved_schedule_cubit.dart]: Ista stvar kao u prethodnom komentaru, samo si emitajte ovako:
`emit(state.copyWith(error: e));`

File: lib/domain/repositories/saved_schedule_repository.dart [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+import '../models/schedule_items.dart';
+
+class SavedRepository {
+  final FirebaseFirestore firestore;
+
+  SavedRepository({required this.firestore});
+
+  Future<List<ScheduleItem>> getSavedItems() async {
+    final docRef = firestore.collection(MyCollections.savedItems).doc(FirebaseAuth.instance.currentUser!.uid);
```
Comment (DreamAgencyGithub) [inline lib/domain/repositories/saved_schedule_repository.dart]: Stavite da `SavedRepository` ima dependency na `FirebaseAuth`

File: lib/domain/repositories/saved_schedule_repository.dart [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+    return savedItemsList;
+  }
+
+  void refreshSavedItems(List<ScheduleItem> items) async {
+    List<String> itemIds = [];
+    for (var element in items) {
+      itemIds.add(element.id);
+    }
+    final docRef = firestore.collection('savedItems').doc(FirebaseAuth.instance.currentUser!.uid);
```
Comment (DreamAgencyGithub) [inline lib/domain/repositories/saved_schedule_repository.dart]: Mislim da u `MyCollections` imate `savedItems` već?

File: lib/domain/repositories/saved_schedule_repository.dart [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+  SavedRepository({required this.firestore});
+
+  Future<List<ScheduleItem>> getSavedItems() async {
+    final docRef = firestore.collection(MyCollections.savedItems).doc(FirebaseAuth.instance.currentUser!.uid);
+    List<dynamic> idList = await docRef.get().then((snapshot) {
+      return snapshot.data()!['savedItems'];
+    });
+
+    DocumentReference<Map<String, dynamic>> savedItemsQuery;
```
Comment (DreamAgencyGithub) [inline lib/domain/repositories/saved_schedule_repository.dart]: Ova varijabla može biti definirana unutar for petlje?

File: lib/domain/repositories/saved_schedule_repository.dart [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    ScheduleItem savedItem;
```
Comment (DreamAgencyGithub) [inline lib/domain/repositories/saved_schedule_repository.dart]: ova također

**Review by DreamAgencyGithub — CHANGES_REQUESTED** [review]

File: lib/application/presentation/personal_schedule.dart (hunk 2 above)
Comment (karlo-rusovan) [reply lib/application/presentation/personal_schedule.dart:14]: fixed, retry button je sada centriran i ne daje exception.

**Pull request merged**